docroot/modules/humsci/hs_config_readonly/src/EventSubscriber/ConfigReadOnlyEventSubscriber.php

```diff
@@ -0,0 +1,222 @@
+      }
+    }
+
+    // Config forms.
+    if ($mark_form_read_only && $form_object instanceof ConfigFormBase) {
+      $editable_config = $this->getEditableConfigNames($form_object);
+      $mark_form_read_only = $this->configIsEditable($editable_config);
+    }
+
+    if ($mark_form_read_only) {
+      $event->markFormReadOnly();
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events = [];
+    $events[ReadOnlyFormEvent::NAME][] = ['onFormAlter', 200];
+    return $events;
+  }
+
+  /**
+   * Get the editable configuration names.
+   *
+   * @param ConfigFormBase $form
+   *   The configuration form.
+   *
+   * @return array
+   *   An array of configuration object names that are editable if called in
+   *   conjunction with the trait's config() method.
+   *
+   * @throws \ReflectionException
+   */
+  protected function getEditableConfigNames(ConfigFormBase $form) {
+    // Use reflection to work around getEditableConfigNames() as protected.
+    // @todo Review in 9.x for API change.
+    // @see https://www.drupal.org/node/2095289
+    $reflection = new \ReflectionMethod(get_class($form), 'getEditableConfigNames');
+    $reflection->setAccessible(TRUE);
+    return $reflection->invoke($form);
+  }
+
+  /**
+   * Check if the given configuration name can be edited.
+   *
+   * @param array $config
+   *   Array of config names.
+   *
+   * @return bool
+   *   If it can be edited.
+   */
+  protected function configIsEditable(array $config) {
+    if ($config == array_filter($config, [$this, 'isEditableConfig'])) {
+      return FALSE;
+    }
+    return TRUE;
+  }
+
+  /**
+   * @param $name
+   *
+   * @return bool
+   */
+  protected function isEditableConfig($name) {
+    foreach ($this->getLockedConfigs() as $config) {
+      if ($config == $name) {
+        return FALSE;
+      }
+    }
+    return TRUE;
+  }
+
+  /**
+   * Get all configs provided by modules.
+   *
+   * @return array
+   *   Config names.
+   */
+  protected function getLockedConfigs() {
+    $module_configs = [];
+    foreach ($this->moduleHandler->getModuleList() as $name => $extension) {
+      if ($extension->getType() == 'profile' || in_array($name, $this->excludedModules)) {
+        continue;
+      }
+
+      $install_list = $this->listProvidedItems($name);
+      $optional_list = $this->listProvidedItems($name, TRUE);
+      $module_configs = array_merge($module_configs, $install_list, $optional_list);
+    }
+    return $module_configs;
+  }
+
+  /**
+   * Returns a list of the install storage items for an extension.
+   *
+   * @param string $name
+   *   Machine name of extension.
+   * @param bool $do_optional
+   *   FALSE (default) to list config/install items, TRUE to list
+   *   config/optional items.
+   *
+   * @return string[]
+   *   List of config items provided by this extension.
+   */
+  protected function listProvidedItems($name, $do_optional = FALSE) {
```

Comment:
I would expect this function by the name to always return the config/install items and return both config/install and config/optional if $do_optional == true. Could you refactor this name to be more clear?

Reply:
its a take over of the over-ridden service. i'm happy to rename and put a comment referencing the method is comes from.

Reply:
i renamed the variable, but it was noted in the comments above the method ;)